src/Bicep.Core/LanguageConstants.cs

```diff
@@ -272,7 +273,23 @@ public static IEnumerable<TypeProperty> CreateResourceProperties(ResourceTypeRef
             {
                 yield return prop;
             }
+            yield return new TypeProperty("location", String);
+
+            yield return new TypeProperty("tags", Tags);
+
+            yield return new TypeProperty("properties", Object);
+
+            foreach (var prop in KnownTopLevelResourceProperties())
+            {
+                yield return prop;
+            }
+
+            var resourceRefArray = new TypedArrayType(ResourceRef, TypeSymbolValidationFlags.Default);
+            yield return new TypeProperty(ResourceDependsOnPropertyName, resourceRefArray, TypePropertyFlags.WriteOnly | TypePropertyFlags.DisallowAny);
+        }
 
+        public static IEnumerable<TypeProperty> KnownTopLevelResourceProperties()
```

Comment:
What's the reason to break this out into different methods? I'd expect the original list of properties in `CreateResourceProperties` to be valid (including location, tags & properties).

Reply:
This is a leftover from initial implementation. I'll merge it back.

Reply:
On the second thought I need to have it separated, so I will know, which properties are fallbacks and which aren't (dependsOn should not have a FallbackProperty flag). I think this method and the one above are used to create a generic resources and removing some elements from here might cause unexpected behaviour, so I left the logic intact but extracted properties on which I need to put the fallback property.
I'd like to leave it as is.

Reply:
Apart from `dependsOn` (see other comment), why would you want to exclude location, tags, and properties from the fallback logic?

Reply:
I don't know. I thought they are always there. Plus, in issue we didn't mention them. But if I think again, they shouldn't be excluded from the logic. 

But the common ones (id, name, type, apiVersion) shouldn't be marked as fallbacks, right? They are also injected in the CreateResourceProperties method:
```
foreach (var prop in GetCommonResourceProperties(resourceTypeReference))
{
    yield return prop;
}
```

Reply:
Good question - you can exclude the common properties as you can depend on them *always* being set on resources. We enforce this during type generation: https://github.com/Azure/bicep-types-az/blob/81c3131898f43c73a43493f5f8bc640fc5bd53b1/src/generator/src/Bicep.TypeGen.Autorest/Processors/ProviderTypeGenerator.cs#L59-L65

Reply:
I'll try it out. perhaps even this entire method should be moved to AzResourceTypeProvider?

Reply:
@anthony-c-martin - I took a look and we need to keep them (name, id, etc.). They are used [here](https://github.com/azure/bicep/blob/main/src/Bicep.Core/TypeSystem/Az/AzResourceTypeProvider.cs#L92-L95) to generate a generic resource type (which was in fact used as a workaround for this issue).